app.rb

```diff
@@ -0,0 +1,114 @@
+require_relative 'book'
+require_relative 'label'
+require_relative 'book_methods'
+require_relative 'music_album'
+require_relative 'music_album_store'
+require_relative 'game_author'
+require_relative 'game'
+require 'json'
+
+class App
+  attr_reader :books, :labels
+
+  def initialize
+    @books = []
+    @labels = []
+  end
+
+  def get_input(prompt, type = :to_s)
+    print prompt
+    input = gets.chomp
+    type == :to_i ? input.to_i : input
+  end
+
+  def list_labels
+    puts 'Listing all labels:'
+    labels.each do |label|
+      puts "Label #{label.id}: #{label.title} (#{label.color})"
+    end
+  end
+
+  def add_label
+    puts 'Enter label title:'
+    title = gets.chomp
+    puts 'Enter label color:'
+    color = gets.chomp
+    label = Label.new(title, color)
+    labels << label
+    label
+  end
+
+  def save_books(filename)
+    File.write(filename, JSON.generate(books.map(&:to_h)))
+  end
+
+  def save_labels(filename)
+    File.write(filename, JSON.generate(labels.map(&:to_h)))
+  end
```

Comment:
- [ ] While I perfectly able to save and also list all books, etc **When I have not exited from the app,**. However, **On exit from the app and reload**, I am not able to list the: Games, Authors, Books, and labels that were there before I closed/ exited the app. KIndly look into why these specific features are not functioning as required.

**Just for some context, below is a screencast where I show practically a before and after.**

[Screencast from 03-04-2023 12:23:45 ALASIRI.webm](https://user-images.githubusercontent.com/86318284/229468855-9294f638-74fe-46c0-bad2-2b06ce229f8e.webm)